### PR TITLE
Implemented aggregation over important mark times.

### PR DESCRIPTION
**Pull request by dgquintas**
Implemented aggregation over important mark times.
    
Namely, 50,90,95 and 99th percentiles are now reported on important marks.
    
Example output:
   
```
Block marks:
============
Block tag          50th p.      90th p.      95th p.      99th p.
201         :        9.789       12.594       14.277       15.508
205         :        9.964       12.849       14.449       15.817
500         :        9.987       15.150       15.814       16.916


Important marks:
================
222222/7f6e6b7f6700@src/core/iomgr/tcp_posix.c:545
Relative mark                                            50th p.      90th p.      95th p.      99th p.
205 { (src/core/iomgr/tcp_posix.c:541)            :        0.020        0.035        0.039        0.050
205 } (src/core/iomgr/tcp_posix.c:557)            :        9.927       12.259       14.400       14.993
500 { (src/core/surface/call.c:1176)              :        1.373        2.367        2.438        2.563
500 } (src/core/surface/call.c:1276)              :       10.510       13.139       15.061       15.764

222222/7f6e6adf4700@src/core/iomgr/tcp_posix.c:545
Relative mark                                            50th p.      90th p.      95th p.      99th p.
500 { (src/core/surface/call.c:1176)              :        1.428        2.401        2.486        2.650
205 { (src/core/iomgr/tcp_posix.c:541)            :        0.025        0.031        0.036        0.045
205 } (src/core/iomgr/tcp_posix.c:557)            :        9.950       14.060       14.462       16.206
500 } (src/core/surface/call.c:1276)              :       10.565       14.776       15.174       17.177
```

### CODE REVIEW

**Comment by dgquintas** [comment]
I've just realized it'd be good to always show {'s before }'s in the output. That's a minor change though.

File: tools/profile_analyzer/profile_analyzer.py [hunk 1]
```diff
@@ -64,6 +65,14 @@ def __init__(self, entry, stack):
   def entry(self):
     return self._entry
 
+  def __hash__(self):
+    # A mark happening at the same file and line will always have the same
+    # stack.
+    return hash('{thread}/{tag}/{type}/{id}/{file}/{line}')
```
Comment (ctiller) [inline tools/profile_analyzer/profile_analyzer.py]: should this have a .format()?

Comment (dgquintas) [inline tools/profile_analyzer/profile_analyzer.py]: woops, yes. I've removed it though, it wasn't actually used.

**Comment by dgquintas** [comment]
Updated code fixing the issue with percentiles of negatives (by getting rid of negatives altogether).

Also improved ordering: { marks are always reported prior to their corresponding } ones.

PTAL.

**Comment by dgquintas** [comment]
Updated for sort().

Another huuuge performance drain is line [123](https://github.com/dgquintas/grpc/blob/profile_analysis_better_reporting/tools/profile_analyzer/profile_analyzer.py#L123). I'll optimize this on a followup changeset.

**Pull request merged**